### ISSUE TITLE
WSL in 22h2 causes total system crash

### ISSUE THREAD

**Issue by gweissman86** [microsoft/WSL#8977]
On a fresh windows install of 22h2 I enabled wsl. On restart, got 2 consecutive BSOD before the computer went into recovery mode and I was able to do a system restore. This is on a Dell XPS laptop. Similar to problem described here (by someone else) https://learn.microsoft.com/answers/questions/1021807/i-can39t-enable-wsl-on-windows-11-version-22h2.html

**Comment by minor-coder** [comment]
I am having the same problem.
Also, running wsl causes intermittent delays in keyboard input, followed by a BSOD before long.

**Comment by minor-coder** [comment]
![8A0815B3-329F-4619-9244-95FDC232C6AC](https://user-images.githubusercontent.com/47618653/194918866-d5750d98-038d-40c9-b5d8-7c13e53a2f8c.jpg)

**Comment by tusharsnn** [comment]
Have you tried [store version](https://www.microsoft.com/store/productId/9P9TQF7MRM4R) once? Just enable virtual-machine-platform feature in the 'windows features'. You can leave wsl disabled there.

**Comment by Biswa96** [comment]
@tusharsnn lxcore.sys is integrated with Windows OS and not part of WSL Preview from store.

**Comment by OneBlue** [comment]
Thank you for reporting this @gweissman86. Can you please follow the [Collect logs via feedback hub section ](https://github.com/Microsoft/WSL/blob/master/CONTRIBUTING.md#9-collect-wsl-logs-with-feedback-hub) and share the feedback hub link here ?

**Comment by gweissman86** [comment]
I made some progress diagnosing this issue. I tried again to enable WSL but this time using the "Turn Windows Features on or off" dialog. This time, the crash only happened when I enabled "Virtual Machine Platform" along with "Windows Subsystem for Linux." So the problem seems to be coming from the Virtual Machine piece of the installation.

**Comment by gweissman86** [comment]
Some more research: The problem seems almost certainly about virtualization, so looks like it's not WSL related. Here is a thread where folks have made some better progress: https://answers.microsoft.com/en-us/insider/forum/insider_wintp-insider_update/windows-11-22h2-getting-stuck/de6fdc6d-2fb9-4e76-b3df-c14cf6c1774e?page=1

**Comment by gweissman86** [comment]
I fixed it on my Dell xps. I had to go into the bios and disable I/O virtualization, and then everything worked again. No idea if ThinkPad has a similar setting, but worth a shot... (I also have no idea what features are lost without I/O virtualization... unfortunately zero real responses from MS or Dell about this.)

**Comment by ducanbk13** [comment]
Hi @gweissman86 , I went to the BIOS and disabled the I/O virtualization, it worked perfectly now. Hope everything works fine without I/O virtualization. Seem to this one is an error from Microsoft, not from the computer hardware.

**Comment by tusharsnn** [comment]
> I fixed it on my Dell xps. I had to go into the bios and disable I/O virtualization, and then everything worked again. No idea if ThinkPad has a similar setting, but worth a shot... (I also have no idea what features are lost without I/O virtualization... unfortunately zero real responses from MS or Dell about this.)

If I/O virtualization is turned off you might not be able to use wsl2 as it creats a linux VM and needs hardware virtualization turned on in the bios.

**Comment by ducanbk13** [comment]
Actually, I just checked again that I disabled both the DMA protection and I/O virtualization. But after I only disabled the  DMA protection without disabling I/O virtualization, the WSL also works well. I think the main problem here is that DMA protection prevents the creation of a new virtual environment. The detail of DMA protection can be referenced here: [https://learn.microsoft.com/en-us/windows/security/information-protection/kernel-dma-protection-for-thunderbolt].
